tests/cloudpickle_test.py

```diff
@@ -2321,6 +2325,21 @@ def __type__(self):
         o = MyClass()
         pickle_depickle(o, protocol=self.protocol)
 
+    def test_deep_serialization_external_func(self):
+        reference = cloudpickle.dumps(an_external_function, protocol=cloudpickle.DEFAULT_PROTOCOL)
+        f1 = cloudpickle.loads(reference)
+
+        register_pickle_by_value("tests.external")
+        deep = cloudpickle.dumps(an_external_function, protocol=cloudpickle.DEFAULT_PROTOCOL)
+        f2 = cloudpickle.loads(deep)
+        unregister_pickle_by_value("tests.external")
+
+        # Ensure the serialisation is not the same
+        assert reference != deep
+        assert len(deep) > len(reference)
```

Comment:
that would be great to have something more explicit. Maybe test that a code object is sent using something like:

```suggestion
        assert b"CodeType" in deep
        assert b"CodeType" not in reference
```

should work.

```

Reply:
or even maybe `types.FunctionType` (may be named `Types.LambdaType`, not 100% sure)

Reply:
I utilised your initial suggest and added this to the tests